CMakeLists.txt

```diff
@@ -21,8 +21,13 @@ if (NOT DEFINED OPENSSL_ROOT_DIR)
     endif ()
 endif ()
 
-set(CMAKE_C_COMPILER "/usr/bin/gcc")
-set(CMAKE_CXX_COMPILER "/usr/bin/g++")
+if (NOT DEFINED CMAKE_C_COMPILER)
+    set(CMAKE_C_COMPILER "/usr/bin/gcc")
+endif ()
+
+if (NOT DEFINED CMAKE_CXX_COMPILER)
+    set(CMAKE_CXX_COMPILER "/usr/bin/g++")
+endif ()
```

Comment:
CMake should default to use `/usr/bin/cc` or `/usr/bin/c++` which will be aliases to `gcc`/`g++` on Linux, so this shouldn't be needed.